subsys/bluetooth/controller/hci/hci_driver.c

```diff
@@ -158,8 +158,8 @@ isoal_status_t sink_sdu_alloc_hci(const struct isoal_sink    *sink_ctx,
 	struct net_buf *buf  = bt_buf_get_rx(BT_BUF_ISO_IN, K_FOREVER);
 
 	if (buf) {
-		/* Reserve space for headers */
-		net_buf_reserve(buf, SDU_HCI_HDR_SIZE);
+		/* Increase reserved space for headers */
+		net_buf_reserve(buf, SDU_HCI_HDR_SIZE + net_buf_headroom(buf));
```

Comment:
Is it correct to have called `net_buf_reserve()` again after is has already been called in `bt_buf_get_rx()` ?

Reply:
`net_buf_reserve` overwrites any previous reserved memory. Perhaps we need a `net_buf_add_reserve`?

Reply:
You are asking whether `buf->user_data` is affected by `net_buf_reserve`? It's not, but the documentation of `net_buf_reserve` is not very clear, and it even says "Initialize buffer".

Note that, confusingly, `net_buf_reserve` replaces the old reservation with the new, it does not add to it.

What do you propose we do? I see two alternatives:

Instead of using `net_buf_reserve`, we can inline it and say exactly what we want:

```c
__ASSERT_NO_MSG(buf->len == 0);
buf->data += SDU_HCI_HDR_SIZE;
```

Or we can use add and pull to achieve the same result.

```c
__ASSERT_NO_MSG(buf->len == 0);
net_buf_add(buf, SDU_HCI_HDR_SIZE);
net_buf_pull(buf, SDU_HCI_HDR_SIZE);
```

Do you have a preference?

Reply:
Indeed. There isn't an API to increase reserved space, only set it. 

> Do you have a preference?

Of the two, I definitely prefer the former. Using `net_buf_pull` to increase reserved size seems way off :D 

There's also my suggestion with `net_buf_add_reserve` (or `net_buf_reserve_add`) we can suggest as a new API for net bufs. I wonder if there are other users doing the same

Reply:
I experimented with the later, and see that the former option,i.e. the current use of `net_buf_reserve`  is functionality the least change required to fix the issue at hand. Approving this PR, and any improvement can be taken outside this PR.